src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlConnectionX.cs

```diff
@@ -53,25 +95,48 @@ public override string Database
         public override int ConnectionTimeout
             => throw new NotImplementedException();
 
-        /// <inheritdoc/>
+        /// <summary>
+        /// Gets or sets the connection string used to connect to the database.
+        /// </summary>
         [AllowNull]
         [DefaultValue("")]
         [RefreshProperties(RefreshProperties.All)]
         [SettingsBindable(true)]
         public override string ConnectionString
         {
-            get => throw new NotImplementedException();
-            set => throw new NotImplementedException();
+            get => _connectionString;
+            set
+            {
+                switch (State)
+                {
+                    case ConnectionState.Open:
+                    case ConnectionState.Connecting:
+                    case ConnectionState.Fetching:
+                    case ConnectionState.Executing:
+                        throw new InvalidOperationException("Can't change connection string on an active connection");
```

Comment:
Nit: exception message should be in a resource string. I think there's already an existing method which will throw the `InvalidOperationException`.

Reply:
Thanks @edwardneal, do you have an example you can point me to? I see some competing patterns in the existing code.

Reply:
The example I'm thinking of is here:
https://github.com/dotnet/SqlClient/blob/55f48c57a9b2a200c266b8739a2a477f974ab41d/src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionHelper.cs#L74-L88